pdfgen.c

```diff
@@ -159,6 +159,7 @@ enum {
     OBJ_catalog,
     OBJ_pages,
     OBJ_image,
+    OBJ_header,
```

Comment:
There's quite a bit in this PR. Can we pull the OBJ_header stuff out for now, and deal with that separately? Just make this PR about the bmp/png support?